src/components/tx/security/shared/SecurityWarnings/index.tsx

```diff
@@ -0,0 +1,173 @@
+                fill: ({ palette }) => palette[severityProps.color].main,
+              },
+            }}
+          />
+        }
+      >
+        {severity !== SecuritySeverity.NONE && <Typography variant="h5">{pluralizedLabel}</Typography>}
+        <Box display="flex" flexDirection="column" gap={2}>
+          {warnings.map((warning) => (
+            <Typography key={warning} variant="body2">
+              {warning}
+            </Typography>
+          ))}
+        </Box>
+      </Alert>
+    </>
+  )
+}
+
+export const SecurityWarning = ({
+  severity,
+  isLoading,
+  error,
+  children,
+  needsConfirmation,
+  isConfirmed,
+  setIsConfirmed,
+}: {
+  severity: SecuritySeverity | undefined
+  isLoading: boolean
+  error: Error | undefined
+  children: ReactElement
+  needsConfirmation: boolean
+  isConfirmed: boolean
+  setIsConfirmed: (value: boolean) => void
+}) => {
+  const isDarkMode = useDarkMode()
+  const severityProps = severity !== undefined ? mapSeverityComponentProps[severity] : undefined
+
+  return (
+    <Box className={css.wrapperBox}>
+      <Accordion className={css.verdictBox}>
+        <AccordionSummary sx={{ mb: 0 }} expandIcon={<ExpandMoreIcon />}>
+          <Grid container direction="row" justifyContent="space-between" alignItems="center">
+            <Grid item>
+              <Typography fontWeight={700} variant="subtitle1">
+                Scan for risks
+              </Typography>
+              <Typography
+                variant="caption"
+                color="text.secondary"
+                display="flex"
+                flexDirection="row"
+                gap={1}
+                alignItems="center"
+                position="relative"
+              >
+                Powered by{' '}
+                <SvgIcon
+                  inheritViewBox
+                  sx={{ height: '40px', width: '52px', position: 'absolute', right: '-20px' }}
+                  component={isDarkMode ? RedefineLogoDark : RedefineLogo}
+                />
+              </Typography>
+            </Grid>
+            {isLoading ? (
+              <LoadingLabel />
+            ) : severityProps ? (
+              <Typography variant="body2" fontWeight={700} color={`${severityProps.color}.main`}>
+                {severityProps.label}
+              </Typography>
+            ) : error ? (
+              <Typography variant="body2" fontWeight={700} color="error">
+                {error.message}
+              </Typography>
+            ) : null}
+          </Grid>
+        </AccordionSummary>
+        <AccordionDetails>{children}</AccordionDetails>
+      </Accordion>
+      {needsConfirmation && (
+        <Box pl={2}>
+          <Track {...MODALS_EVENTS.ACCEPT_RISK}>
+            <FormControlLabel
+              label="I understand the risks and would like to continue this transaction"
+              control={<Checkbox checked={isConfirmed} onChange={() => setIsConfirmed(!isConfirmed)} />}
```

Comment:
```suggestion
              control={<Checkbox checked={isConfirmed} onChange={() => setIsConfirmed((prev) => !prev)} />}
```

Reply:
This only makes sense if you also memoize the function…